### ISSUE TITLE
Inference Token Inclusion and SFT Question Loss Calculation Queries

### ISSUE THREAD

**Issue by ruio248** [simplescaling/s1#72]
**Question:**  
I am using the "budget force" method for inference. Should I include the `<|im_start|>answer` token during the final inference step in order to match the training format of the `s1.1_tokenized` dataset?

**Question:**  
It seems that the SFT training script includes a calculation of the loss for the `question`. Could you explain the benefit of doing this?

**Comment by Muennighoff** [comment]
1) if you want the model to stop early then you can insert `<|im_start|>answer`; if you want it to think longer, you can ignore it
2) Loss over the question is not calculated I think; only over the reasoning trace & answer
